### DISCUSSION TITLE
Remote imgcat in terminal

### DISCUSSION THREAD

**Discussion by hhoeflin** [wezterm/wezterm#1009] [Q&A]
Hi,

I like the imgcat feature in wezterm. I installed wezterm on my windows machine and ssh-ed into a linux machine, but then the terminal img display did not work. How can I get this to work? Do I have to install wezterm also on the remote machine? Is there another way to handle this that is "lighter weight"?

Thanks

**Comment by kar1** [comment]
imgcat encodes images in terminal control codes for the terminal to display. On the remote site you need another application to do the encoding if wezterm is not installed. You can try: https://github.com/hzeller/timg and run it on the remote side. fx:

`timg -C -pk image_name.png`

You can change the display method with the `-p` flag. From `timg -h`:

```
        -p<pixelation> : Pixelation: 'h' = half blocks    'q' = quarter blocks
                                     'k' = kitty graphics 'i' = iTerm2 graphics
                         Default: Auto-detect graphics, otherwise 'quarter'.
```

`timg` should detect the terminal, and if so, only: `timg image.png` is needed.

As an alternative, you can use iTerm2's imgcat script here: https://iterm2.com/utilities/imgcat

-- edited
And you should check out https://notcurses.com/ it's really cool. A modern curses lib with user tools. It's the future for tui IMHO

**Reply by SuperSandro2000** [reply]
That does not work through tmux for me.

**Reply by hhoeflin** [reply]
Yeah the author mentions something about tmux issues with that in the docs.

**Reply by kar1** [reply]
tmux acts like a middleman between the shell and the terminal. So if tmux doesn't let the terminal codes pass through it won't work. I believe tmux insist on encapsulating the, for tmux, unrecognized terminal codes. But I'm not sure.

**Reply by kar1** [reply]
FWIW There is a discussion on the issue here: https://gitlab.com/gnachman/iterm2/-/issues/3898

**Comment by bew** [comment]
FWIW I've made my own shell-based implementation of imgcat, based on a script provided by iterm2 which @kar1 mentioned (which is actually incomplete wrt the actual spec):
https://github.com/bew/dotfiles/blob/main/bin/imgcat
(has a help if run without args)

It does kinda work through tmux, with limitations (because tmux doesn't understand the img protocol):
- scrolling a pane will hide the img
- if run from a prompt, you will need to add a few blank lines after the imgcat cmd to leave room for the img before the next shell prompt (limitation of tmux). I've made a [separate `blank-lines` helper](https://github.com/bew/dotfiles/blob/main/bin/blank-lines) for this, I use it like `imgcat  --height 12 foo.png; blank-lines 12` (I might later add the echo loop in the imgcat script when run under tmux)
